Formula/g/gap.rb

```diff
@@ -0,0 +1,38 @@
+class Gap < Formula
+  desc "Groups, Algorithms, Programming - A system for computational discrete algebra"
+  homepage "https://www.gap-system.org"
+  url "https://github.com/gap-system/gap/releases/download/v4.13.0/gap-4.13.0.tar.gz"
+  sha256 "cc76ecbe33d6719450a593e613fb87e9e4247faa876f632dd0f97c398f92265d"
+  license "GPL-2.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "libtool" => :build
+  depends_on "gmp"
+  depends_on :macos
+  depends_on "readline"
+  depends_on :xcode
+  uses_from_macos "zlib"
+
+  def install
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--with-gmp=#{Formula["gmp"].include}",
+      "--with-readline=#{Formula["readline"].include}"
+    system "make", "install"
+    pkgshare.install "pkg"
+  end
+
+  test do
+    system bin/"gap", "--version"
```

Comment:
If we do a version check, let's see if it works:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/gap --version")
```

Reply:
Removed version check. The functionality test is valid and the tar has a hashsum - not really an important test.